addons/account/account_move_line.py

```diff
@@ -917,6 +917,24 @@ def reconcile_partial(self, cr, uid, ids, type='auto', context=None, writeoff_ac
         move_rec_obj.reconcile_partial_check(cr, uid, [r_id] + merges_rec, context=reconcile_context)
         return r_id
 
+    def _prepare_writeoff_line_vals(self, cr, uid, ids, name, debit, credit,
+                                    account_id, date, analytic_account_id,
+                                    cur_id, amount_currency, partner_id,
+                                    currency, account_currency_id,
+                                    context=None):
+        """This method is designed to be inherited in a custom module"""
+        return {'name': name,
+                'debit': debit,
+                'credit': credit,
+                'account_id': account_id,
+                'date': date,
+                'analytic_account_id': analytic_account_id,
+                'partner_id': partner_id,
+                'currency_id': cur_id or (account_currency_id or False),
+                'amount_currency': amount_currency and -1 * amount_currency or
+                (account_currency_id and -1 * currency or 0.0)
```

Comment:
This doesn't seem correct: multiplying by -1 should not be done for writeoff_acc_id